pkg/kube/translation/ingress.go

```diff
@@ -87,29 +94,49 @@ func (t *translator) translateIngressV1(ing *networkingv1.Ingress) (*TranslateCo
 				ctx.addUpstream(ups)
 			}
 			uris := []string{pathRule.Path}
-			if pathRule.PathType != nil && *pathRule.PathType == networkingv1.PathTypePrefix {
-				// As per the specification of Ingress path matching rule:
-				// if the last element of the path is a substring of the
-				// last element in request path, it is not a match, e.g. /foo/bar
-				// matches /foo/bar/baz, but does not match /foo/barbaz.
-				// While in APISIX, /foo/bar matches both /foo/bar/baz and
-				// /foo/barbaz.
-				// In order to be conformant with Ingress specification, here
-				// we create two paths here, the first is the path itself
-				// (exact match), the other is path + "/*" (prefix match).
-				prefix := pathRule.Path
-				if strings.HasSuffix(prefix, "/") {
-					prefix += "*"
-				} else {
-					prefix += "/*"
+			var nginxVars []configv2alpha1.ApisixRouteHTTPMatchExpr
+			if pathRule.PathType != nil {
+				if *pathRule.PathType == networkingv1.PathTypePrefix {
+					// As per the specification of Ingress path matching rule:
+					// if the last element of the path is a substring of the
+					// last element in request path, it is not a match, e.g. /foo/bar
+					// matches /foo/bar/baz, but does not match /foo/barbaz.
+					// While in APISIX, /foo/bar matches both /foo/bar/baz and
+					// /foo/barbaz.
+					// In order to be conformant with Ingress specification, here
+					// we create two paths here, the first is the path itself
+					// (exact match), the other is path + "/*" (prefix match).
+					prefix := pathRule.Path
+					if strings.HasSuffix(prefix, "/") {
+						prefix += "*"
+					} else {
+						prefix += "/*"
+					}
+					uris = append(uris, prefix)
+				} else if *pathRule.PathType == networkingv1.PathTypeImplementationSpecific && useRegex {
+					nginxVars = append(nginxVars, configv2alpha1.ApisixRouteHTTPMatchExpr{
+						Subject: configv2alpha1.ApisixRouteHTTPMatchExprSubject{
+							Scope: configv2alpha1.ScopePath,
+						},
+						Op:    configv2alpha1.OpRegexMatch,
+						Value: &pathRule.Path,
```

Comment:
I think we should validate the regex pattern in the admission server.

Reply:
this path comes from ingress resources, I don't thinke we need to do this.

Reply:
Why? Does the Ingress resource validator validate it?

Reply:
> 

Even it comes from the Ingress, we still have to validate it before translating, or it cannot be used by APISIX.

Reply:
currently, the apisix-ingress-controller will not validate all the ingress resource being created? Maybe valiate the ingress resource should be another feature.

On the other hand, a issue in kubernetes projects https://github.com/kubernetes/kubernetes/issues/90210
the validation was removed

Reply:
All right, that'd be unexpected, would you like to submit an issue? I think we can implement it in the future.

Reply:
ok